src/components/views/right_panel/HeaderButton.tsx

```diff
@@ -19,18 +19,33 @@ limitations under the License.
 */
 
 import React from 'react';
-import PropTypes from 'prop-types';
 import classNames from 'classnames';
 import Analytics from '../../../Analytics';
 import AccessibleTooltipButton from "../elements/AccessibleTooltipButton";
 
-export default class HeaderButton extends React.Component {
-    constructor() {
-        super();
+interface IProps {
+    // Whether this button is highlighted
+    isHighlighted: boolean;
+    // click handler
+    onClick: () => void;
+    // The badge to display above the icon
+    badge?: React.ReactNode;
+    // The parameters to track the click event
+    analytics: string[];
+
+    // Button name
+    name: string;
+    // Button title
+    title: string;
+}
+
+export default class HeaderButton extends React.Component<IProps> {
+    constructor(props: IProps) {
+        super(props);
         this.onClick = this.onClick.bind(this);
     }
 
-    onClick(ev) {
+    onClick(_ev: React.KeyboardEvent) {
```

Comment:
We don't really do the '_' thing

Reply:
It's not important though

Reply:
The `ev` parameter is not being used, so I wanted to make that explicit.

Reply:
You could just omit the parameter

Reply:
I didn't want to change the type signature, but it seems that's alright to do. Done